CONTRIBUTING.md

```diff
@@ -2,192 +2,176 @@
 
-Note that some features are disabled by default for local builds.
+    ```console
+    git clone --recursive https://github.com/cp-algorithms/cp-algorithms.git && cd cp-algorithms
+    ```
 
-### Git revision date plugin
+2. Install dependencies and serve the site:
 
-Disabled because it might produce errors when there are uncommitted changes in the working tree.
+    ```console
+    $ scripts/install-mkdocs.sh # requires pip
+    $ mkdocs serve
+    ```
 
-To enable it, set the environment variable `MKDOCS_ENABLE_GIT_REVISION_DATE` to `True`:
+   This will run the site locally so you can preview your changes. Note that some features are disabled in local builds.
 
-```console
-$ export MKDOCS_ENABLE_GIT_REVISION_DATE=True
-```
+### Optional Plugins
 
-### Git committers plugin
+- **Git Revision Date Plugin**: Disabled by default but can be enabled with:
 
-Disabled because it takes a while to prepare and also requires Github personal access token to work with Github APIs.
+    ```console
+    $ export MKDOCS_ENABLE_GIT_REVISION_DATE=True
+    ```
 
-To enable it, set the environment variable `MKDOCS_ENABLE_GIT_COMMITTERS` to `True` and store your personal access token in the environment variable `MKDOCS_GIT_COMMITTERS_APIKEY`. You can generate the token [here](https://github.com/settings/tokens). Note that you only need the public access, so you shouldn't give the token any permissions.
+- **Git Committers Plugin**: Disabled by default, as it requires a GitHub personal access token. Enable it like this:
 
-```console
-$ export MKDOCS_ENABLE_GIT_COMMITTERS=True
-$ export MKDOCS_GIT_COMMITTERS_APIKEY= # put your PAT here 
-```
+    ```console
+    $ export MKDOCS_ENABLE_GIT_COMMITTERS=True
+    $ export MKDOCS_GIT_COMMITTERS_APIKEY=your_token_here
+    ```
 
-## Tests
+   You can generate your token [here](https://github.com/settings/tokens). Only public access permissions are needed.
 
-If your article involves code snippets, then it would be great you also contribute tests for them.
-This way we can make sure that the snippets actually work, and don't contain any bugs.
+## Testing Code Snippets
 
-Creating tests works like this:
-You have to name each snippet that you want to test in the markdown article:
+If your article includes code snippets, it’s helpful to include tests to ensure that they run correctly.
 
+1. Name the code snippet:
     ```{.cpp file=snippet-name}
-    // some code
+    // code here
     ```
+2. Run `extract_snippets.py` from the `test` directory to extract snippets into header files. Create a test file that includes these headers and checks their behavior.
+3. You can run all tests with the `test.sh` script:
+    ```console
+    $ cd test
+    $ ./test.sh
+    ```
+    **Example Output:**
+    ```
+    Running test_aho_corasick.cpp - Passed in 635 ms
+   Running test_balanced_brackets.cpp - Passed in 1390 ms
+    Running test_burnside_tori.cpp - Passed in 378 ms
+    ...
+    51 PASSED in 49.00 seconds
+    ```
```

Comment:
```suggestion
    ```
    Running test_aho_corasick.cpp - Passed in 635 ms
    Running test_balanced_brackets.cpp - Passed in 1390 ms
    Running test_burnside_tori.cpp - Passed in 378 ms
    ...
    51 PASSED in 49.00 seconds
    ```
```
Otherwise it misrenders.